extensions/blocks/mailchimp/view.js

```diff
@@ -15,12 +15,16 @@ import './view.scss';
 
 const blockClassName = 'wp-block-jetpack-mailchimp';
 
-function fetchSubscription( blogId, email ) {
-	const url =
+function fetchSubscription( blogId, email, params ) {
+	let url =
 		'https://public-api.wordpress.com/rest/v1.1/sites/' +
 		encodeURIComponent( blogId ) +
 		'/email_follow/subscribe?email=' +
 		encodeURIComponent( email );
+
+	for ( const param in params ) {
+		url += '&' + param + '=' + params[ param ];
```

Comment:
Would using `addQueryArgs` from `@wordpress/url` add too much to view-side bundle? It's just so much more ergonomic. 😁 Totes understand if it's too many kbs tho so not a blocker.

Reply:
This is also an interesting one, and I think it'll work since we're polyfilling blocks: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
In @artpi's original original version of the block, the API call is a `GET` just like it is here. When we added AMP support `amp-form` requires the method be POST so we now have two pretty different approaches to submitting the form. Ideally, I'd like to switch the non-AMP form to POST as well to simplify things, I'm just wary of doing that at this moment since it will introduce new complexity and muddy up this PR. Because of this I'm inclined not to introduce `addQueryArgs` since that will be a short-lived solution anyway, and it's really just a code clarity improvement. Would that be ok with you?

Reply:
That totally makes sense! :thumbsup: Thanks for elaborating.

Reply:
I'll open an issue for this and look into it in a future PR.